pandas/io/parsers.py

```diff
@@ -2327,14 +2327,21 @@ def _next_line(self):
                 try:
                     orig_line = next(self.data)
                 except csv.Error as e:
+                    msg = str(e)
+
                     if 'NULL byte' in str(e):
-                        raise csv.Error(
-                            'NULL byte detected. This byte '
-                            'cannot be processed in Python\'s '
-                            'native csv library at the moment, '
-                            'so please pass in engine=\'c\' instead.')
-                    else:
-                        raise
+                        msg = ('NULL byte detected. This byte '
+                               'cannot be processed in Python\'s '
+                               'native csv library at the moment, '
+                               'so please pass in engine=\'c\' instead')
+
+                    if self.skipfooter > 0:
+                        reason = ('Error could possibly be due to '
+                                  'skipfooter being ignored in Python\'s '
+                                  'native csv library.')
```

Comment:
Yes, Python's native CSV library does not care about `skipfooter`, so any malformed lines in the `skipfooter` section will raise an error.  However, if there are no such errors, these rows are then discarded afterwards.